src/downloader.rs

```diff
@@ -105,25 +105,25 @@ impl Downloader {
         match req.send() {
             Ok(mut data) => {
                 let data_type = match data.headers().get("content-type") {
-                    Some(data_type) => data_type.to_str().unwrap(),
-                    None => "text/html",
+                    Some(data_type) => data_type.to_str().unwrap().to_string(),
+                    None => String::from("text/html"),
                 };
 
-                let filename = if !Downloader::is_html(data_type) {
+                let filename = if !Downloader::is_html(&data_type) {
                     Downloader::get_filename(data.headers())
                 } else {
                     None
                 };
 
-                let data = if Downloader::is_html(data_type) {
-                    ResponseData::Html(data.text().unwrap())
+                let mut raw_data: Vec<u8> = Vec::new();
+                data.copy_to(&mut raw_data).unwrap();
```

Comment:
I don't really mind the copy if it is necessary, but can't we use a `&[u8]` instead? And maybe have a method like `data.as_bytes()` (Maybe it already exists?)

If ownership is necessary or too complex then disregard this comment

Reply:
That's not possible because data was created in this function. If we took a reference to ``data.bytes().XXX`` to get a ``&[u8]``, we will put it in a ``ResponseData`` which in turn will be put inside a ``Response`` and this will leak outside the function inside the ``Ok()``

Reply:
Okay perfect :+1: As long as we know it's not possible